unity-client/Assets/Scripts/MainScripts/DCL/ServiceProviders/Catalyst/Catalyst.cs

```diff
@@ -0,0 +1,218 @@
+        }
+
+        Promise<string>[] splittedPromises = new Promise<string>[pointersGroupsToFetch.Length];
+        
+        for (int i = 0; i < pointersGroupsToFetch.Length; i++)
+        {
+            string urlParams = "";
+            urlParams = pointersGroupsToFetch[i].Aggregate(urlParams, (current, pointer) => current + $"&pointer={pointer}");
+            string url = $"{realmDomain}/content/entities/{entityType}?{urlParams}";
+            
+            splittedPromises[i] = Get(url);
+            splittedPromises[i].Then(value =>
+            {
+                // check if all other promises have been resolved
+                for (int j = 0; j < splittedPromises.Length; j++)
+                {
+                    if (splittedPromises[j] == null || splittedPromises[j].keepWaiting || !string.IsNullOrEmpty(splittedPromises[j].error))
+                    {
+                        return;
+                    }
+                }
+
+                // make sure not to continue if promise was already resolved
+                if (!promise.keepWaiting)
+                    return;
+                
+                // build json with all promises result
+                string json = splittedPromises[0].value.Substring(1, splittedPromises[0].value.Length - 2);
+                for (int j = 1; j < splittedPromises.Length; j++)
+                {
+                    string jsonContent = splittedPromises[j].value.Substring(1, splittedPromises[j].value.Length - 2);
+                    json += $",{jsonContent}";
+                }
+                promise.Resolve($"[{json}]");
+            });
+            splittedPromises[i].Catch(error => promise.Reject(error));
+        }
+
+        return promise;
+    }
+
+    public Promise<string> Get(string url)
+    {
+        Promise<string> promise = new Promise<string>();
+
+        if (cache.TryGetValue(url, out string cachedResult))
+        {
+            promise.Resolve(cachedResult);
+            return promise;
+        }
+
+        WebRequestController.i.Get(url, null ,request =>
+        {
+            AddToCache(url, request.downloadHandler.text);
+            promise.Resolve(request.downloadHandler.text);
+        }, request =>
+        {
+            promise.Reject($"{request.error} {request.downloadHandler.text} at url {url}");
+        });
+
+        return promise;
+    }
+
+    private void PlayerRealmOnOnChange(CurrentRealmModel current, CurrentRealmModel previous)
+    {
+        realmDomain = current.domain;
+        realmContentServerUrl = current.contentServerUrl;
+    }
+
+    private void AddToCache(string url, string result)
+    {
+        cache[url] = result;
+
+        // NOTE: remove from cache after CACHE_TIME time passed
+        CoroutineStarter.Start(RemoveCache(url, CACHE_TIME));
+    }
+
+    private IEnumerator RemoveCache(string url, float delay)
+    {
+        yield return WaitForSecondsCache.Get(delay);
+        cache?.Remove(url);
+    }
```

Comment:
Would it be more perfomance to just check time at the moment of do the call to check if the cache has expire? This way we wouldn't have a courutine running

Reply:
I thought about that, but my main concern was that the result json from the catalyst can be huge and I didn't want it to stay stored at the dictionary permanently.